groups/bdl/bdlb/bdlb_guidutil.cpp

```diff
@@ -137,6 +197,39 @@ void GuidUtil::generate(Guid *result, bsl::size_t numGuids)
     generate(reinterpret_cast<unsigned char *>(result), numGuids);
 }
 
+void GuidUtil::generateNonSecure(unsigned char *result, bsl::size_t numGuids)
+{
+    unsigned char *bytes = result;
+    unsigned char *end   = bytes + numGuids * Guid::k_GUID_NUM_BYTES;
+
+    static bdlb::PCG pcgSingleton;
+    BSLMT_ONCE_DO
+    {
+        uint64_t seed;
+        if (0 != RandomDevice::getRandomBytes((unsigned char *)&seed,
+                                              sizeof(seed))) {
+            seed = time(NULL) ^ (intptr_t)&bsl::printf;  // fallback seed
+        }
+        int rounds = 5;  // see PCG code base for details of 'rounds'
+        pcgSingleton.seed(seed, (intptr_t)&rounds);
```

Comment:
... and here we construct a `static bdlb::PCG pcgSingleton(seed, streamSelector);` and then set the pointer. 

Also, please adjust this initialization because `(intptr_t)&rounds` doesn't make much sense. I'd expect the stream selector to also come from `getRandomBytes`.